exist-core/src/main/java/org/exist/xquery/functions/fn/transform/Convert.java

```diff
@@ -119,11 +125,18 @@ static net.sf.saxon.s9api.QName of(final QNameValue qName) {
         }
 
         XdmValue of(final Item item) throws XPathException {
+            if (item instanceof NodeProxy) {
+              return ofNode(((NodeProxy) item).getNode());
+            }
```

Comment:
As of Java 21 you could use this:

```suggestion
            if (item instanceof NodeProxy nodeProxy) {
              return ofNode(nodeProxy.getNode());
            }
```

Reply:
this PR is to 6.x.x